onnxruntime/core/session/inference_session.cc

```diff
@@ -1505,6 +1505,114 @@ common::Status InferenceSession::ValidateOutputs(const std::vector<std::string>&
   return common::Status::OK();
 }
 
+#ifdef ENABLE_TRAINING
+Status InferenceSession::PartialRun(onnxruntime::RunOptions& run_options, const std::vector<OrtValue>& feeds, std::vector<OrtValue>& fetches,
+                                    PartialGraphExecutionState& state, FeedsFetchesManager& feeds_fetches_manager) {
+  TimePoint tp;
+  if (session_profiler_.IsEnabled()) {
+    tp = session_profiler_.Now();
+  }
+
+#ifdef ONNXRUNTIME_ENABLE_INSTRUMENT
```

Comment:
maybe we can skip the telemetry at all, before we figuring out how to do instrumentation for training.

Reply:
trying to get those last bits of performance I see :P haha ...

Reply:
@mrry @KeDengMS Curious what your take is on this?

Reply:
Sherlock makes a good point - I don't think we'll ever exercise this code path, and since we don't make training builds with `-DONNXRUNTIME_ENABLE_INSTRUMENT` it's effectively dead code. 

However, I think we'll eventually - in a subsequent PR - want to find a refactoring of this code and `InferenceSession::Run()` that avoids most of the repetition, and we'll need to take the union of functionality, so we'll probably end up leaving it in :).

Reply:
Removed telemetry from training.